pkg/api/v1alpha2/api.proto

```diff
@@ -0,0 +1,487 @@
+        option (google.api.http) = {
+            post: "/api/Manager/ValidateAlgorithmSettings"
+            body: "*"
+        };
+    };
+}
+
+service Suggestion {
+    rpc GetSuggestions(GetSuggestionsRequest) returns (GetSuggestionsReply);
+    rpc ValidateAlgorithmSettings(ValidateAlgorithmSettingsRequest) returns (ValidateAlgorithmSettingsReply);
+}
+
+service EarlyStopping {
+//TODO
+}
+
+/**
+ * Types of value for HyperParameter.
+ */
+enum ParameterType {
+    UNKNOWN_TYPE = 0; /// Undefined type and not used.
+    DOUBLE = 1; /// Double float type. Use "Max/Min".
+    INT = 2; /// Int type. Use "Max/Min".
+    DISCRETE = 3; /// Discrete number type. Use "List" as float.
+    CATEGORICAL = 4; /// Categorical type. Use "List" as string.
+}
+
+
+/**
+ * Feasible space for optimization.
+ * Int and Double type use Max/Min.
+ * Discrete and Categorical type use List.
+ */
+message FeasibleSpace {
+    string max = 1; /// Max Value
+    string min = 2; /// Minimum Value
+    repeated string list = 3; /// List of Values.
+    string step = 4; /// Step for double or int parameter
+}
+
+/**
+ * Config for a Hyper parameter.
+ * Katib will create each Hyper parameter from this config.
+ */
+message ParameterSpec {
+    string name = 1; /// Name of the parameter.
+    ParameterType parameter_type = 2; /// Type of the parameter.
+    FeasibleSpace feasible_space = 3; /// FeasibleSpace for the parameter.
+}
+
+/**
+ * Direction of optimization. Minimize or Maximize.
+ */
+enum ObjectiveType {
+    UNKNOWN = 0; /// Undefined type and not used.
+    MINIMIZE = 1; /// Minimize
+    MAXIMIZE = 2; /// Maximize
+}
+
+message ObjectiveSpec{
+    ObjectiveType type = 1;
+    float goal = 2;
+    string objective_metric_name = 3;
+    repeated string additional_metrics_names = 4; 
+    /// This can be empty if we only care about the objective metric.
+}
+
+message AlgorithmSetting {
+    string name = 1;
+    string value = 2;
+}
+
+message EarlyStoppingSpec {
+//TODO
+}
+
+message AlgorithmSpec {
+    string algorithm_name = 1;
+    repeated AlgorithmSetting algorithm_setting = 2;
+    EarlyStoppingSpec early_stopping_spec = 3;
+}
+
+/**
+ * TODO add to AlgorithmSpec
+ * NasConfig contains a config of NAS job
+ */
+ message NasConfig {
+    GraphConfig graph_config = 1; /// Config of DAG
+    message Operations {
+        repeated Operation operation = 1;
+    }
+    Operations operations = 2; /// List of Operation
+}
+
+/**
+ * GraphConfig contains a config of DAG
+ */
+ message GraphConfig {
+    int32 num_layers = 1; /// Number of layers
+    repeated int32 input_size = 2; /// Dimensions of input size
+    repeated int32 output_size = 3; /// Dimensions of output size
+}
+
+/**
+ * Config for operations in DAG
+ */
+ message Operation {
+     string operation_type = 1; /// Type of operation in DAG
+      /** 
+     * List of ParameterAssignment
+     */
+     message ParameterAssignments {
```

Comment:
@YujiOshima @richardsliu I think we need `ParameterSpec` instead of `ParameterAssignments`, that I told you. We need Feasible space inside ParameterConfigs in Operations.

Reply:
OK, I will update.

Reply:
Thank you!